### PR TITLE
LTBENCH-157: Add an option to zip final report

### PR DESCRIPTION
**Pull request by vilmantas-r**

### CODE REVIEW

File: test-automation-allure/src/main/kotlin/com/nortal/test/report/allure/services/AllureReportZipService.kt [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+import java.util.zip.ZipOutputStream
+
+/**
+ * Compress Allure report to zip file.
+ */
+@Component
+class AllureReportZipService {
+
+    @Throws(IOException::class)
+    fun zipReport(reportDir: String, zipOutputDir: String) {
```
Comment (ricardas-buc) [inline test-automation-allure/src/main/kotlin/com/nortal/test/report/allure/services/AllureReportZipService.kt:40]: use Kotlin "use" function. Something like:
`fun zipReport(reportDir: String, zipOutputDir: String) {
    val reportToZip = File(reportDir)

    val zipName = String.format("%s.zip", reportToZip.name)
    val zipFile = File(zipOutputDir, zipName)
    zipFile.parentFile.mkdirs()

    FileOutputStream(zipFile).use { fos ->
        ZipOutputStream(fos).use { zipOut ->
            zipFile(reportToZip, reportToZip.name, zipOut)
        }
    }
}`

**Review by ricardas-buc — CHANGES_REQUESTED** [review]

File: test-automation-allure/src/main/kotlin/com/nortal/test/report/allure/services/AllureReportZipService.kt (hunk 1 above)
Comment (vilmantas-r) [reply test-automation-allure/src/main/kotlin/com/nortal/test/report/allure/services/AllureReportZipService.kt:40]: Refactored

File: test-automation-allure/src/main/kotlin/com/nortal/test/report/allure/services/AllureReportZipService.kt [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        val zipName = String.format("%s.zip", reportToZip.name)
+        val zipFile = File(zipOutputDir, zipName)
+        zipFile.parentFile.mkdirs()
+
+        FileOutputStream(zipFile).use { fos ->
+            ZipOutputStream(fos).use { zipOut ->
+                zipFile(reportToZip, reportToZip.name, zipOut)
+                zipOut.close()
+            }
+            fos.close()
```
Comment (ricardas-buc) [inline test-automation-allure/src/main/kotlin/com/nortal/test/report/allure/services/AllureReportZipService.kt]: the benefit of 'use' is that you do not need to close it. It is closed automatically.

**Review by ricardas-buc — CHANGES_REQUESTED** [review]

File: test-automation-allure/src/main/kotlin/com/nortal/test/report/allure/services/AllureReportZipService.kt [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+                zipOut.putNextEntry(ZipEntry("$fileName/"))
+                zipOut.closeEntry()
+            }
+            val children = fileToZip.listFiles()
+            for (childFile in children!!) {
+                zipFile(childFile, fileName + "/" + childFile.name, zipOut)
+            }
+            return
+        }
+        val fis = FileInputStream(fileToZip)
```
Comment (ricardas-buc) [inline test-automation-allure/src/main/kotlin/com/nortal/test/report/allure/services/AllureReportZipService.kt]: Another stream that can be refactored in same manner

File: test-automation-allure/src/main/kotlin/com/nortal/test/report/allure/services/AllureReportZipService.kt (hunk 2 above)
Comment (vilmantas-r) [reply test-automation-allure/src/main/kotlin/com/nortal/test/report/allure/services/AllureReportZipService.kt]: Fixed

**Review by ricardas-buc — CHANGES_REQUESTED** [review]

File: test-automation-allure/src/main/kotlin/com/nortal/test/report/allure/services/AllureReportZipService.kt (hunk 3 above)
Comment (vilmantas-r) [reply test-automation-allure/src/main/kotlin/com/nortal/test/report/allure/services/AllureReportZipService.kt]: Refactored

File: test-automation-allure/src/main/resources/application-base.yml [hunk 4]
```diff
@@ -3,4 +3,7 @@ test-automation:
     allure:
       base-dir: build
       report-dir: build/allure-report
-      result-dir: build/allure-results
\ No newline at end of file
+      result-dir: build/allure-results
+      zip-report:
+        enabled: true
```
Comment (ricardas-buc) [inline test-automation-allure/src/main/resources/application-base.yml]: disable by default. It can be set to true in tests, but not in main source.

**Review by ricardas-buc — APPROVED** [review]

**Pull request merged**